pynbody/filt.py

```diff
@@ -100,36 +100,50 @@ class Sphere(Filter):
     def __init__(self, radius, cen=(0, 0, 0)):
         self._descriptor = "sphere"
         self.cen = np.asarray(cen)
-        if self.cen.shape != (3,):
```

Comment:
Removing this check could lead to a segfault in _sphere_selection

You could implement it as self.cen.shape[-1]!=3?